### ISSUE TITLE
Customerize loss function

### ISSUE THREAD

**Issue by shyTNT** [Microsoft/CNTK#2891]
hi, I'd like to define my own loss function based on cross_entropy_with_softmax. But I am not sure if I use it correctly. 

Here is the code example. Is that correct? Or are there any examples for loss function with cntk ops? Thanks.

Before:
**ce = cross_entropy_with_softmax(m, labels)**
After:
**ce = create_cost_function(z, m, labels)**

def create_cost_function(secondlastlayer, lastlayer, prediction):
         ces = cross_entropy_with_softmax(lastlayer, prediction)
         new_loss= new_loss_func(secondlastlayer, lastlayer, prediction)
         return  ces + new_loss

def new_loss_func(secondlastlayer, lastlayer, prediction):
         sqrVector = cntk.sqrt(cntk.reduce_sum(cntk.pow(secondlastlayer, 2)))
         sqrWeight = cntk.sqrt(cntk.reduce_sum(cntk.pow(prediction, 2)))
         return sqrVector * sqrWeight

**Comment by KeDengMS** [comment]
You can try different kinds of loss function with the richness in CNTK's expression. The script looks correct to me from syntax point of view, but it's up to you to decide whether it's suitable for your model.

**Comment by shyTNT** [comment]
Yes, I just want to make sure the syntax and whether or not it works. Thanks.
